joint_trajectory_controller/CMakeLists.txt

```diff
@@ -44,22 +50,38 @@ target_include_directories(joint_trajectory_controller PUBLIC
 )
 target_link_libraries(joint_trajectory_controller PUBLIC
   joint_trajectory_controller_parameters
+  ruckig::ruckig
 )
 ament_target_dependencies(joint_trajectory_controller PUBLIC ${THIS_PACKAGE_INCLUDE_DEPENDS})
 
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.
 target_compile_definitions(joint_trajectory_controller PRIVATE "JOINT_TRAJECTORY_CONTROLLER_BUILDING_DLL" "_USE_MATH_DEFINES")
+
+add_library(cartesian_trajectory_generator SHARED
+  src/cartesian_trajectory_generator.cpp
+)
+target_compile_features(cartesian_trajectory_generator PUBLIC cxx_std_17)
+target_include_directories(cartesian_trajectory_generator PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include/joint_trajectory_controller>
+)
+target_link_libraries(cartesian_trajectory_generator PUBLIC
+  joint_trajectory_controller
+)
+ament_target_dependencies(cartesian_trajectory_generator PUBLIC ${THIS_PACKAGE_INCLUDE_DEPENDS})
+
 pluginlib_export_plugin_description_file(controller_interface joint_trajectory_plugin.xml)
 
 if(BUILD_TESTING)
   find_package(ament_cmake_gmock REQUIRED)
   find_package(controller_manager REQUIRED)
   find_package(ros2_control_test_assets REQUIRED)
 
-  ament_add_gmock(test_trajectory test/test_trajectory.cpp)
-  target_link_libraries(test_trajectory joint_trajectory_controller)
-  target_compile_definitions(test_trajectory PRIVATE _USE_MATH_DEFINES)
+  # TODO(henrygerardmoore): re-enable
+  # ament_add_gmock(test_trajectory test/test_trajectory.cpp)
+  # target_link_libraries(test_trajectory joint_trajectory_controller)
+  # target_compile_definitions(test_trajectory PRIVATE _USE_MATH_DEFINES)
```

Comment:
Why is this commented out?

Reply:
Let me see if it can be re-enabled. If I remember correctly, it was making the build fail.

Reply:
Ah yes, the test is using the old version of trajectory.sample. There are a lot of them, so I thought I would leave re-enabling it until later (since the fix is more involved than just a find and replace, too). Is that okay?

Reply:
Did we change trajectory.sample in our fork? Then it is ok as it will get fixed when we merge upstream?

Reply:
Yes, we changed that in our fork and it should be fixed when we merge upstream